packages/@aws-cdk/aws-servicecatalogappregistry/lib/application-associator.ts

```diff
@@ -29,12 +41,81 @@ export interface ApplicationAssociatorProps {
   readonly description?: string;
 
   /**
-   * Stack properties.
+   * Stack ID to which application will be created or imported.
+   *
+   * @default - ApplicationAssociatorStack
+   */
+  readonly stackId?: string;
+
+  /**
+   * Stack properties to which Application will be created or imported.
    *
    */
   readonly stackProps: cdk.StackProps;
 }
 
+/**
+ * Class which constructs the input from provided Application ARN.
+ */
+class ImportApplicationProps implements IBaseApplicationAssociatorProps {
```

Comment:
Here's a pattern to follow for these classes, so you can put the behavior INSIDE the individual case classes (instead of in the main class which works by inspecting certain properties):

```ts
export abstract class ApplicationTarget {
  public static existingApplication(...) {
    return new ExistingApplicationTarget(...);
  }
 
  public static createApplication(...) {
    return new CreateApplicationTarget(...);
  } 

  public abstract bind(scope: Construct, options: BindApplicationTargetOptions): BindApplicationTargetResult;
}

export interface BindApplicationTargetOptions { /* if you need to pass options into the class */ }

export interface BindApplicationTargetResult {
  readonly applicationArn: string;
  /* any other output you need to make decisions on */
}

/* note: no export */ class CreateApplicationTarget extends ApplicationTarget {
  constructor(...) {
    super();
  }

  public bind(scope: Construct, options: BindApplicationTargetOptions): BindApplicationTargetResult {
    const stack = new Stack(scope, '...', { ... });
    new Application(stack, '...', { ... });

    return {
      applicationArn: 'arn:aws:...:application/MyName',
    };
  }
}
```

Put them in a separate file as well please.